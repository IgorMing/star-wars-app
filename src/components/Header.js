import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Text,
  StyleSheet,
  Platform,
  Image,
  StatusBar,
  TouchableWithoutFeedback,
  View
} from 'react-native';
import { SimpleLineIcons } from '@expo/vector-icons';

export default class Header extends Component {
  static propTypes = {
    leftAction: PropTypes.func,
    rightIcon: PropTypes.string
  };

  static defaultProps = {
    leftAction: () => {},
    rightIcon: ''
  };

  renderLeft() {
    const { leftAction } = this.props;

    if (!leftAction) {
      return null;
    }

    return (
      <View style={styles.headerLeft}>
        <TouchableWithoutFeedback onPress={leftAction}>
          <SimpleLineIcons name="logout" size={20} color="white" />
        </TouchableWithoutFeedback>
      </View>
    );
  }

  renderRightIcon() {
    const { rightIcon } = this.props;

    if (!rightIcon) {
      return null;
    }

    return (
      <View>
        <Image
          style={styles.picture}
          source={{ uri: rightIcon }}
        />
      </View>
    );
  }

  render() {
    return (
      <View style={styles.header}>
        {this.renderLeft()}
        <View style={styles.header__title}>
          <Text style={styles.header__title__text}>Star wars app</Text>
        </View>
        {this.renderRightIcon()}
      </View>
    );
  }
}

const paddingVertical = 10;
const horizontalDistance = 18;
const statusBarHeight = Platform.OS === 'ios' ? 0 : StatusBar.currentHeight;

const styles = StyleSheet.flatten({
  header: {
    paddingTop: statusBarHeight + paddingVertical,
    paddingBottom: paddingVertical,
    paddingLeft: horizontalDistance,
    flexDirection: 'row',
    backgroundColor: '#222'
  },
  header__title: {
    flex: 9,
    justifyContent: 'center',
    marginLeft: horizontalDistance
  },
  header__title__text: {
    color: 'white',
    fontSize: 17
  },
  headerLeft: {
    flex: 1,
    justifyContent: 'center'
  },
  picture: {
    width: 45,
    height: 45,
    borderRadius: 50,
    marginRight: horizontalDistance
  }
});
