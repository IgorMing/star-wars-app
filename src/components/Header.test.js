import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

describe('Header component tests', () => {
  const mockedImage = 'http://geekycentral.com/wp-content/uploads/2017/09/react-native.png';
  const mockedFn = jest.fn();

  it('should render without crashing', () => {
    const wrapper = renderer.create(<Header />).toJSON();

    expect(wrapper).toBeTruthy();
  });

  it('should render with right icon', () => {
    const wrapper = renderer
      .create(<Header rightIcon={mockedImage} />)
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a left action and without right icon', () => {
    const wrapper = renderer
      .create(<Header leftAction={mockedFn} />)
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a left action and with right icon', () => {
    const wrapper = renderer
      .create(<Header leftAction={mockedFn} rightIcon={mockedImage} />)
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});

