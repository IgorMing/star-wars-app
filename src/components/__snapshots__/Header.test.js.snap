// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests should render with a left action and with right icon 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#222",
      "flexDirection": "row",
      "paddingBottom": 10,
      "paddingLeft": 18,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
      testID={undefined}
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 9,
        "justifyContent": "center",
        "marginLeft": 18,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 17,
        }
      }
    >
      Star wars app
    </Text>
  </View>
  <View>
    <Image
      source={
        Object {
          "uri": "http://geekycentral.com/wp-content/uploads/2017/09/react-native.png",
        }
      }
      style={
        Object {
          "borderRadius": 50,
          "height": 45,
          "marginRight": 18,
          "width": 45,
        }
      }
    />
  </View>
</View>
`;

exports[`Header component tests should render with a left action and without right icon 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#222",
      "flexDirection": "row",
      "paddingBottom": 10,
      "paddingLeft": 18,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
      testID={undefined}
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 9,
        "justifyContent": "center",
        "marginLeft": 18,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 17,
        }
      }
    >
      Star wars app
    </Text>
  </View>
</View>
`;

exports[`Header component tests should render with right icon 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#222",
      "flexDirection": "row",
      "paddingBottom": 10,
      "paddingLeft": 18,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
      testID={undefined}
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 9,
        "justifyContent": "center",
        "marginLeft": 18,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 17,
        }
      }
    >
      Star wars app
    </Text>
  </View>
  <View>
    <Image
      source={
        Object {
          "uri": "http://geekycentral.com/wp-content/uploads/2017/09/react-native.png",
        }
      }
      style={
        Object {
          "borderRadius": 50,
          "height": 45,
          "marginRight": 18,
          "width": 45,
        }
      }
    />
  </View>
</View>
`;
